src/source/RazorEngine.Core/Templating/IsolatedRazorEngineService.cs

```diff
@@ -87,7 +87,16 @@ public DefaultConfigCreator()
             /// <returns></returns>
             public ITemplateServiceConfiguration CreateConfiguration()
             {
-                return new TemplateServiceConfiguration();
+                // We need to use the DefaultCompilerServiceFactory because we
+                // get conflicting TypeLoadExceptions in RazorEngineSourceReferenceResolver
+                // that cannot be resolved.
+                // ie) When not used inside the IsolatedRazorEngineService, we
+                // need [SecuritySafeCritical] for the methods. However, inside
+                // the Isolated service, it requires everything to be [SecurityCritical].
```

Comment:
I don't get that part completely, is it maybe because the Roslyn dependency is not marked as SecurityCritical?

I guess I have to take a look at this.